### PR TITLE
feat: deploy with docker

### PR DESCRIPTION
**Pull request by dont-see-big-shark**
deploy with docker

### CODE REVIEW

**Review by miaojuncn — APPROVED** [review]

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: '3'
```
Comment (rainchen) [inline docker-compose.yaml]: file name should be docker-compose.yml

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+services:
+  cita-ide:
+    build:
+      context: .
+      dockerfile: ./Dockerfile
+    image: cita-ide:latest
+    container_name: cita-ide
+    ports:
+      - 9000:9000
```
Comment (rainchen) [inline docker-compose.yaml]: should keep a newline at end of file

File: nginx.conf [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    charset utf-8;
+    client_max_body_size 10M;
+    add_header 'Access-Control-Allow-Credentials' 'true';
+	  add_header 'Access-Control-Allow-Origin' '*';
+    location / {
+        root /usr/share/nginx/html;
+        index index.html;
+        try_files $uri $uri/ /index.html;
+    }
+}
```
Comment (rainchen) [inline nginx.conf]: should keep a newline at end of file

File: nginx.conf [hunk 4]
```diff
@@ -0,0 +1,12 @@
+server {
```
Comment (rainchen) [inline nginx.conf]: should move this  to a `docker/nginx/` dir, same as other projects

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,13 @@
+FROM node:10 as build
+WORKDIR /app
+COPY . .
```
Comment (rainchen) [inline Dockerfile:5]: should add COPY package.json and put COPY . . after npm build command, check the best practice for build docker image for a nodejs project

**Review by rainchen — CHANGES_REQUESTED** [review]
please update

**Review by rainchen — APPROVED** [review]

**Pull request merged**